### PR TITLE
KRAK-278 add selectable enable sysdig agent to cloudinit

### PR DESCRIPTION
**Pull request by mikeln**
KRAK-278  Add sysdig cloud agent in ansible such that if a sysdig key is present the service is included and started.  Else, the service is not included.

### CODE REVIEW

**Comment by mikeln** [comment]
hmm.. hold off, the rebase tests are not correct yet.

**Comment by mikeln** [comment]
should be good now

**Comment by maratoid** [comment]
retest this please

**Comment by maratoid** [comment]
retest this please

File: terraform/aws/variables.tf [hunk 1]
```diff
@@ -15,6 +15,11 @@ variable "aws_user_prefix" {
   description = "AWS resource prefix - all resources with names will be identified as <aws_user_prefix>_<cluster_name>_resource. Also used in kubectl --user"
 }
 
+variable "sysdigcloud_access_key" {
+  description = "Sysdig Cloud Access Key"
+  default = "DISABLED"
```
Comment (maratoid) [inline terraform/aws/variables.tf]: can we just use a "" for no token?

File: ansible/roles/kubernetes/templates/sysdig-cloud-agent.service.ansible [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+[Service]
+Type=simple
+Restart=on-failure
+RestartSec=5
+TimeoutStartSec=5
+ExecStartPre=-/usr/bin/docker kill sysdig-agent
+ExecStartPre=-/usr/bin/docker rm sysdig-agent
+ExecStartPre=/usr/bin/docker pull sysdig/agent
+ExecStart=/usr/bin/docker run --name sysdig-agent --privileged --net host --pid host -e ACCESS_KEY={{sysdigcloud_access_key}} -v /var/run/docker.sock:/host/var/run/docker.sock -v /dev:/host/dev -v /proc:/host/proc:ro -v /boot:/host/boot:ro sysdig/agent
```
Comment (spiffxp) [inline ansible/roles/kubernetes/templates/sysdig-cloud-agent.service.ansible]: can you break this up into multiple lines?  see cadvisor.service.ansible for an example

File: terraform/aws/variables.tf (hunk 1 above)
Comment (mikeln) [inline terraform/aws/variables.tf]: we could.  I really prefer the explicit value so the intention is clear.

Comment (maratoid) [inline terraform/aws/variables.tf]: I don't feel strongly for either way, but we should at least do it one way in all cases.

Comment (mikeln) [inline terraform/aws/variables.tf]: Hesitant to say, I do have a strong preference on this.
We just need make sure we have a ticket to make consistent.

Comment (spiffxp) [inline terraform/aws/variables.tf]: We're using the `""` convention most everywhere else, could we please stick to that

File: ansible/roles/kubernetes/templates/sysdig-cloud-agent.service.ansible (hunk 2 above)
Comment (mikeln) [inline ansible/roles/kubernetes/templates/sysdig-cloud-agent.service.ansible]: done

File: terraform/aws/variables.tf (hunk 1 above)
Comment (mikeln) [inline terraform/aws/variables.tf]: Hmm. ok under protest.

**Comment by spiffxp** [comment]
lgtm, waiting for jenkins to have its say

**Pull request merged**